src/groupByKey.ts

```diff
@@ -2,6 +2,18 @@ type GroupsMap<T> = {
   [key: string]: T[];
 };
 
-export function groupByKey(items, key) {
-  // write code here;
+export function groupByKey<T>(items: T[], key: keyof T): GroupsMap<T> {
+  const result: GroupsMap<T> = {};
+
+  items.forEach((current) => {
+    const nKey = current[key] as keyof GroupsMap<T>;
+
+    if (result[nKey] === undefined) {
+      result[nKey] = [current];
+    } else {
+      result[nKey].push(current);
+    }
+  }, {});
```

Comment:
what is empty object doing here?

Reply:
Oh, yeah I forgot about this object because firstly I wrote this by reduce method.